### 유방암 진단 데이터를 분류합시다.

__1.__ 자료를 데이터 분석하기에 적합한 dataframe으로 불러옵니다.

In [18]:
import pandas as pd
df = pd.read_csv('breast-cancer-wisconsin.csv')

__2.__ dataframe을 탐색적 분석합니다.

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 683 entries, 0 to 682
Data columns (total 11 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   code                         683 non-null    int64
 1   Clump_Thickness              683 non-null    int64
 2   Cell_Size                    683 non-null    int64
 3   Cell_Shape                   683 non-null    int64
 4   Marginal_Adhesion            683 non-null    int64
 5   Single_Epithelial_Cell_Size  683 non-null    int64
 6   Bare_Nuclei                  683 non-null    int64
 7   Bland_Chromatin              683 non-null    int64
 8   Normal_Nucleoli              683 non-null    int64
 9   Mitoses                      683 non-null    int64
 10  Class                        683 non-null    int64
dtypes: int64(11)
memory usage: 58.8 KB


|index     |column|
|----------|-------|
|0 to 682  |11개  |

|column name | data type|
|------------|----------|
|code|int|
|

In [20]:
df.head()

code  Clump_Thickness  Cell_Size  Cell_Shape  Marginal_Adhesion  \
0  1000025                5          1           1                  1   
1  1002945                5          4           4                  5   
2  1015425                3          1           1                  1   
3  1016277                6          8           8                  1   
4  1017023                4          1           1                  3   

   Single_Epithelial_Cell_Size  Bare_Nuclei  Bland_Chromatin  Normal_Nucleoli  \
0                            2            1                3                1   
1                            7           10                3                2   
2                            2            2                3                1   
3                            3            4                3                7   
4                            2            1                3                1   

   Mitoses  Class  
0        1      0  
1        1      0  
2        1      0  
3        1      0  
4        1      0

In [21]:
df['Class'] = df['Class'].replace([0,1],["정상","유방암환자"])

In [22]:
df['Class'].value_counts()

정상       444
유방암환자    239
Name: Class, dtype: int64

__3.__ 데이터셋을 학습데이터셋( <font color="#B451E8"> __feature__</font> )과 테스트데이터셋( <font color="#B451E8"> __test__</font> )으로 나눕니다.
* 컬럼 이름으로 나누는 방법
* 컬럼 인덱스로 나누는 방법
* loc함수로 나누는 방법

In [ ]:
x1 = data[['Clump_thickness']]